osf_tests/test_region.py

```diff
@@ -0,0 +1,68 @@
+import mock
+import pytest
+
+from framework.auth import Auth
+
+from osf.models import MetaSchema, Registration
+from osf_tests.factories import (
+    AuthUserFactory,
+    ProjectFactory,
+    RegionFactory
+)
+
+
+@pytest.mark.django_db
+class TestRegion:
+
+    @pytest.fixture()
+    def user(self):
+        return AuthUserFactory()
+
+    @pytest.fixture()
+    def project_with_different_regions(self, user):
+        """
+        A complex project configuration with many regions.
+        :param user:
+        :return:
+        """
+        parent_node = root_node = ProjectFactory(creator=user)
+
+        # components nested five deep
+        for _ in range(0, 5):
```

Comment:
Is it necessary to create such a large hierarchy? This could make the test slow and hard to debug.